# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/promise/Promise.h
# Commit: ac548c2a725b
# Full Hash: ac548c2a725b66db57c81bb24accca2bbe075931
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2019-07-29 21:38:39
# Description:
#   Bug 1565930, try to make Promise less error prone to compartment mismatches, r=bzbarsky
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D38762
# ==============================================================================

diff -r c4980c083852 -r ac548c2a725b dom/promise/Promise.h
--- a/dom/promise/Promise.h	Mon Jul 29 08:00:35 2019 +0000
+++ b/dom/promise/Promise.h	Mon Jul 29 14:43:54 2019 +0000
@@ -75,9 +75,6 @@
   typedef void (Promise::*MaybeFunc)(JSContext* aCx,
                                      JS::Handle<JS::Value> aValue);
 
-  void MaybeResolve(JSContext* aCx, JS::Handle<JS::Value> aValue);
-  void MaybeReject(JSContext* aCx, JS::Handle<JS::Value> aValue);
-
   // Helpers for using Promise from C++.
   // Most DOM objects are handled already.  To add a new type T, add a
   // ToJSValue overload in ToJSValue.h.
@@ -89,6 +86,9 @@
 
   void MaybeResolveWithUndefined();
 
+  void MaybeReject(JS::Handle<JS::Value> aValue) {
+    MaybeSomething(aValue, &Promise::MaybeReject);
+  }
   inline void MaybeReject(nsresult aArg) {
     MOZ_ASSERT(NS_FAILED(aArg));
     MaybeSomething(aArg, &Promise::MaybeReject);
@@ -236,6 +236,9 @@
                          eDontPropagateUserInteraction);
 
  private:
+  void MaybeResolve(JSContext* aCx, JS::Handle<JS::Value> aValue);
+  void MaybeReject(JSContext* aCx, JS::Handle<JS::Value> aValue);
+
   template <typename T>
   void MaybeSomething(T&& aArgument, MaybeFunc aFunc) {
     MOZ_ASSERT(PromiseObj());  // It was preserved!